daemon/commit.go

```diff
@@ -20,7 +21,10 @@ type ContainerCommitConfig struct {
 // The image can optionally be tagged into a repository.
 func (daemon *Daemon) Commit(container *Container, c *ContainerCommitConfig) (*image.Image, error) {
 	if c.Pause && !container.isPaused() {
-		container.pause()
+		if err := container.pause(); err != nil && err != derr.ErrorCodeNotRunning.WithArgs(container.ID) && err != derr.ErrorCodeAlreadyPaused.WithArgs(container.ID) {
```

Comment:
I'm not thrilled that there isn't a util func in errcode to make this easier, I would prefer if you could just write something like:
```
if ... derr.ErrorCodeNotRunning.isA(err) ...
```
to test to see if 'err' is of type ErrorCodeNotRunning, because doing a strcmp like this (while probably ok) just feel kind of ugly to me.  But that's not your fault.  If you want to add a util func in the derr package as part of this I wouldn't mind  :-)

But, in the meantime, I think the more proper way to do this is to try to cast 'err' to be of type 'ErrorCoder`, and if it is, then just check the ErrorCode() of err and make sure its the same as `derr.ErrorCodeNotRunning`.  Does that make sense?